File: src/Domain/Models/Challenges/RequestsResults/ChallengeSave.cs
using Domain.Entities.Challenges;
using Domain.Entities.Common;

namespace Domain.Models.Challenges.RequestsResults;

public class ChallengeSaveRequest
{
    public string Name { get; set; }

    public string DescriptionShort { get; set; }

    public string DescriptionMarkdown { get; set; }

    public string StubGeneratorInput { get; set; }

    public List<TestPair> Tests { get; set; }

    public Solution Solution { get; set; }

    public List<string> TagIds { get; set; }
}

File: src/react-frontend/src/app/api/challenge/types/challengeSave.ts
import { TestPair } from 'app/types/models/challenge/testPair';
import { Solution } from 'app/types/models/general/solution';

type ChallengeSaveRequest = {
  name: string;
  descriptionShort: string;
  descriptionMarkdown: string;
  stubGeneratorInput: string;
  tests: TestPair[];
  solution: Solution;
  tagIds: string[];
};

export type ChallengeSaveRequestWithParameters = {
  id?: string;
  model?: ChallengeSaveRequest;
};
